PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -440,55 +459,123 @@ struct lambdaCorrTableProducer {
 
-    for (auto mcpart : mcparts) {
-      if (mcpart.globalIndex() != 0) {
-        IDs.push_back(mcpart.globalIndex());
-        PDGs.push_back(mcpart.pdgCode());
-      }
+    // check for lambda/anti-lambda
+    if (abs(mcpart.pdgCode()) != kLambda0) {
+      histos.fill(HIST("Tracks/h1f_tracks_info"), 10.5);
+      return false;
+    }
+
+    // check for primary/secondary lambda
+    if (cfg_rec_primary_lambda && !mcpart.isPhysicalPrimary()) {
+      histos.fill(HIST("Tracks/h1f_tracks_info"), 11.5);
+      return false;
+    } else if (!cfg_rec_primary_lambda && mcpart.isPhysicalPrimary()) {
+      histos.fill(HIST("Tracks/h1f_tracks_info"), 12.5);
+      return false;
+    }
+
+    auto postrack = v0.template posTrack_as<T>();
+    auto negtrack = v0.template negTrack_as<T>();
+
+    // check if the daughters have corresponding mcparticle
+    if (!postrack.has_mcParticle() || !negtrack.has_mcParticle()) {
+      histos.fill(HIST("Tracks/h1f_tracks_info"), 13.5);
+      return false;
+    }
+
+    auto mcpostrack = postrack.template mcParticle_as<aod::McParticles>();
+    auto mcnegtrack = negtrack.template mcParticle_as<aod::McParticles>();
+
+    if (abs(mcpostrack.pdgCode()) != kProton || abs(mcnegtrack.pdgCode()) != kPiPlus) { // incorrect decay channel -> return false
+      histos.fill(HIST("Tracks/h1f_tracks_info"), 14.5);
+      return false;
+    }
+
+    return true;
+  }
+
+  template <ParticleType part, typename C, typename V>
+  float get_corr_factor(C const& col, V const& v0)
+  {
+    // Check for efficiency correction flag and Rec/Gen Data
+    if (!cfg_corr_flag) {
+      return 1.;
     }
 
-    while (IDs.size() > 2) {
-      IDs.pop_back();
+    // Get  from CCDB
+    auto ccdb_obj = ccdb->getForTimeStamp<TList>(cfg_ccdb_path.value, -1);
+
+    // Check CCDB Object
+    if (!ccdb_obj) {
+      LOGF(warning, "CCDB OBJECT NOT FOUND");
+      return 1.;
     }
 
-    while (PDGs.size() > 2) {
-      PDGs.pop_back();
+    // get ccdb object
+    TObject* obj = reinterpret_cast<TObject*>(ccdb_obj->FindObject(Form("%s", v_cf_str[cfg_cf_hist_type][part].c_str())));
+
+    if (obj->InheritsFrom("TH1F")) {
+      TH1F* hist = reinterpret_cast<TH1F*> obj->Clone();
+      int pt_bin = hist->GetXaxis()->FindBin(v0.pt());
+      return hist->GetBinContent(pt_bin);
+    } else if (obj->InheritsFrom("TH2F")) {
+      TH2F* hist = reinterpret_cast<TH2F*> obj->Clone();
+      int pt_bin = hist->GetXaxis()->FindBin(v0.pt());
+      int rap_bin = hist->GetYaxis()->FindBin(v0.yLambda());
+      return hist->GetBinContent(pt_bin, rap_bin);
+    } else if (obj->InheritsFrom("TH3F")) {
+      TH3F* hist = reinterpret_cast<TH3F*> obj->Clone();
+      int pt_bin = hist->GetXaxis()->FindBin(v0.pt());
+      int rap_bin = hist->GetYaxis()->FindBin(v0.yLambda());
+      int vz_bin = hist->GetZaxis()->FindBin(col.posZ());
+      return hist->GetBinContent(pt_bin, rap_bin, vz_bin);
+    } else {
+      return 1.;
     }
   }
```

Comment:
Please, run it locally before pushing the functionality
There are histograms that are cloned on a per V0 basis and that are left there, allocated, which creates memory leaks

Reply:
Dear Victor, I will keep this in mind. I missed compiling the task after the last change, it will not happen again.
I understand the thing about memory leak. I will rectify it in next PR. I am not going to use CCDB object this time because I want to get the correction factors with this task. I added the ccdb method for future purposes. I will keep the memory leak thing in mind and thank you for your suggestion.